File: src/test/java/com/backend/immilog/notification/applicaiton/service/DiscordSendingServiceTest.java
package com.backend.immilog.notification.applicaiton.service;

import com.backend.immilog.notification.applicaiton.command.DiscordCommand;
import com.backend.immilog.notification.infrastructure.gateway.discord.DiscordGateway;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.mockito.ArgumentCaptor;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

@DisplayName("DiscordSendingService 클래스")
class DiscordSendingServiceTest {

    private final DiscordGateway discordGateway = mock(DiscordGateway.class);
    private final DiscordSendingService discordSendingService = new DiscordSendingService(discordGateway);

    @Test
    @DisplayName("디스코드 메시지 전송 성공")
    void sendMethodSendsCorrectDiscordRequest() {
        String api = "testApi";
        Exception exception = new Exception("testException");
        discordSendingService.send(api, exception);
        ArgumentCaptor<DiscordCommand> captor = ArgumentCaptor.forClass(DiscordCommand.class);
        verify(discordGateway).send(captor.capture());
        DiscordCommand capturedRequest = captor.getValue();
        assertThat(capturedRequest.content()).isEqualTo("[EXTERNAL API] {testApi}");
        assertThat(capturedRequest.embeds().getFirst().fields().getFirst().value()).isEqualTo("testException");
    }
}

File: src/test/java/com/backend/immilog/notification/application/service/DiscordSendingServiceTest.kt
package com.backend.immilog.notification.application.service

import com.backend.immilog.notification.application.command.DiscordCommand
import com.backend.immilog.notification.infrastructure.gateway.discord.DiscordGateway
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.mockito.Mockito.*

@DisplayName("DiscordSendingService 클래스")
class DiscordSendingServiceTest {

    private val discordGateway: DiscordGateway = mock(DiscordGateway::class.java)
    private val discordSendingService = DiscordSendingService(discordGateway)

    @Test
    @DisplayName("디스코드 메시지 전송 성공")
    fun sendMethodSendsCorrectDiscordRequest() {
        // Given
        val api = "testApi"
        val exception = Exception("testException")

        // When
        discordSendingService.send(api, exception)

        val expectedRequest = DiscordCommand(
            content = "[EXTERNAL API] {testApi}",
            embeds = listOf(
                DiscordCommand.Embed(
                    title = "Exception Detail",
                    fields = listOf(
                        DiscordCommand.Embed.Field("java.lang.Exception", "testException", true)
                    )
                )
            )
        )

        verify(discordGateway).send(expectedRequest)
    }
}
